llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -676,6 +676,11 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(
 
         if (EarlyInlineAll && !EnableFunctionCalls)
           PM.addPass(AMDGPUAlwaysInlinePass());
+
+#if __has_feature(address_sanitizer)
+        EnableLowerModuleLDS = false;
```

Comment:
You should not be changing the values of global options

Reply:
With ASAN enabled, #87265 will now lower the LDS. I want to disable "lower-module-lds" pass when lowering is done by the new pass. 
Could you please suggest any other way of disabling the "lower-module-lds" pass instead of setting EnableLowerModuleLDS  to false?